src/AppInstallerRepositoryCore/Manifest/YamlManifestParser.cpp

```diff
@@ -0,0 +1,425 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "YamlManifestParser.h"
+
+namespace AppInstaller::Manifest
+{
+    namespace
+    {
+        std::vector<Manifest::string_t> SplitMultiValueField(const std::string& input)
+        {
+            if (input.empty())
+            {
+                return {};
+            }
+
+            std::vector<Manifest::string_t> result;
+            size_t currentPos = 0;
+            while (currentPos < input.size())
+            {
+                size_t splitPos = input.find(',', currentPos);
+                if (splitPos == std::string::npos)
+                {
+                    splitPos = input.size();
+                }
+
+                std::string splitVal = input.substr(currentPos, splitPos - currentPos);
+                Utility::Trim(splitVal);
+                if (!splitVal.empty())
+                {
+                    result.emplace_back(std::move(splitVal));
+                }
+                currentPos = splitPos + 1;
+            }
+
+            return result;
+        }
+    }
+
+    void YamlManifestParser::PrepareManifestFieldInfos(const ManifestVer& manifestVer)
+    {
+        RootFieldInfos =
+        {
+            { "ManifestVersion", PreviewManifestVersion, [this](const YAML::Node&) { /* ManifestVersion already processed */ }, false,
+            // Regex here is to prevent leading 0s in the version, this also keeps consistent with other versions in the manifest

```

Comment:
>Regex [](start = 15, length = 5)

Moving the regex seems to be too much.  If we are separating the parser, it should be to have it parse the syntax of the format, and then we should add these semantics onto that.

Parser > Raw Manifest > Validation

Either that or the parser gets an interface to expose selection of individual values out via some syntax we define, and the top level parser gets passed a syntactic parser to provide that.

Basically, if we are separating the code out, it should be written to enable any syntax.

Reply:
Chatted offline and leave it for now.

---
In reply to: [454031238](https://github.com/microsoft/winget-cli/pull/493#discussion_r454031238) [](ancestors = 454031238)